mitodl/open-discussions
# Added caching for refresh and access tokens

#### What are the relevant tickets?
Fixes #307 

#### What's this PR do?
Adds tables to cache refresh and access tokens. There is a one-to-one mapping of users to refresh tokens and a one-to-many mapping of users to access tokens. This is intended because while the refresh tokens live forever, the access tokens are short-lived (1 hour) and are subject to the thundering heard problem, so we need to be sure not to trample on other access tokens.

The token with the future-most expiration date is always used. Expired tokens are evicted via cron.

#### How should this be manually tested?
- Login to the OD app
- Verify you can use the app and that you now have a `RedditAccessToken` and `RedditRefreshToken` for your user.
- Set the `token_expires_at` date for your token in the past, verify you see a new one generated.
- Run `channels.tasks.evict_expired_access_tokens.delay()` and verify your expired token disappears
